### PR TITLE
Feature/unr 1526 expose replication limit

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Expose replication consider list and rate limit to the spatial metrics.
Also expose modifying GDK settings from clients.

#### Release note
TODO

#### Tests
Testing in TestSuite

#### Documentation
There should be documentation here, but waiting for team signoff first

#### Primary reviewers
@improbable-valentyn @mattyoung-improbable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h [hunk 1]
```diff
@@ -137,18 +137,23 @@ namespace SpatialConstants
...
+
+	// ModifySettingPayload Field IDs
+	const Schema_FieldId MODIFY_SETTING_PAYLOAD_NAME_ID						= 1;
+	const Schema_FieldId MODIFY_SETTING_PAYLOAD_VALUE_ID					= 2;
 
 	// UnrealRPCPayload Field IDs
-	const Schema_FieldId UNREAL_RPC_PAYLOAD_OFFSET_ID = 1;
-	const Schema_FieldId UNREAL_RPC_PAYLOAD_RPC_INDEX_ID = 2;
-	const Schema_FieldId UNREAL_RPC_PAYLOAD_RPC_PAYLOAD_ID = 3;
+	const Schema_FieldId UNREAL_RPC_PAYLOAD_OFFSET_ID						= 1;
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h]: ew

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h]: consistency!

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 2]
```diff
@@ -197,6 +198,73 @@ void USpatialMetrics::OnStopRPCMetricsCommand()
...
+		Worker_EntityId ControllerEntityId = PCObjectRef.Entity;
+
+		if (ControllerEntityId != SpatialConstants::INVALID_ENTITY_ID)
+		{
+			Worker_CommandRequest Request = {};
+			Request.component_id = SpatialConstants::DEBUG_METRICS_COMPONENT_ID;
+			Request.schema_type = Schema_CreateCommandRequest(SpatialConstants::DEBUG_METRICS_COMPONENT_ID, SpatialConstants::DEBUG_METRICS_MODIFY_SETTINGS_ID);
+			
+			Schema_Object* RequestObject = Schema_GetCommandRequestObject(Request.schema_type);
+			SpatialGDK::AddStringToSchema(RequestObject, 1, Name);
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp]: Nit: Add field IDs as constants.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp [hunk 3]
```diff
@@ -197,6 +198,73 @@ void USpatialMetrics::OnStopRPCMetricsCommand()
...
+			Schema_AddFloat(RequestObject, 2, Value);
+
+			NetDriver->Connection->SendCommandRequest(ControllerEntityId, &Request, SpatialConstants::DEBUG_METRICS_MODIFY_SETTINGS_ID);
+		}
+		else
+		{
+			UE_LOG(LogSpatialMetrics, Warning, TEXT("SpatialModifySetting: Could not resolve local PlayerController entity! RPC metrics will not stop on the server."));
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp]: Fix log message.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetricsDisplay.cpp [hunk 4]
```diff
@@ -200,6 +204,8 @@ void ASpatialMetricsDisplay::Tick(float DeltaSeconds)
 	FWorkerStats Stats{};
 	Stats.WorkerName = SpatialNetDriver->Connection->GetWorkerId().Left(WorkerNameMaxLength).ToLower();
 	Stats.AverageFPS = Metrics.GetAverageFPS();
+	Stats.ServerConsiderListSize = SpatialNetDriver->GetConsiderListSize();
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetricsDisplay.cpp]: Is there a `#if !UE_BUILD_SHIPPING` around this? I can't seem to find it.

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetricsDisplay.cpp]: There is not - good catch

**Pull request merged**